contents/security/passwordStorage.md

```diff
@@ -0,0 +1,113 @@
+
+## What is hashing?
+
+Hashing is the act of transforming a set of data into another set of data that is impossible to reverse. Additionally, due to how <trigger for="pop:hashing-algo">hashing algorithms</trigger> work, even a small change in the data will result in a completely different hash. This means that even very similar passwords that vary only by capitalization or a letter would still result in different <tooltip content="Hashes are the result of a hashing function">hashes</tooltip>, so attackers can't compare the hash to try and figure out which users have similar passwords.
+
+<popover id="pop:hashing-algo" title="A _hashing algorithm_ is a specific type of operation that hashes the input" placement="top">
+  <div slot="content">
+Different types of hashing algorithms will result in different output.
+  </div>
+</popover>
+
+Some examples of cryptographic hashing algorithms are [MD5](https://www.quora.com/How-does-the-MD5-algorithm-work) and [SHA1](https://deadhacker.com/2006/02/21/sha-1-illustrated/). However, when you perform hashing for passwords, you should use password hashing algorithms (such as [Argon2](https://github.com/P-H-C/phc-winner-argon2), [SCrypt](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.scrypt.html) and [bcrypt](https://security.stackexchange.com/questions/4781/do-any-security-experts-recommend-bcrypt-for-password-storage)). The main difference between these password hashing algorithms are designed to be slow to prevent <trigger for="pop:brute">brute force</trigger> attacks, unlike cryptographic hashing algorithms which are built for speed. Despite that, if you plan on learning more about hashing algorithms, we recommend starting with MD5 and SHA1, as they are easier algorithms to learn about.
+
+### Why isn't hashing enough?
+
+Because [rainbow tables](https://en.wikipedia.org/wiki/Rainbow_table) exist, unfortunately.
+
+A rainbow table is a precomputed table of hashes for some set of passwords. Basically, since the hashing function is easy to compute one-way, people build huge tables of hashes wherein the plaintext is already known, so that attempting to crack hashes becomes a simple problem of looking up the hash in the table and its corresponding value, instead of attempting to reverse the hash. Because of this, it is very easy to crack simple hashes by simply doing a lookup.
+
+An example of a service that provides this is [Crackstation](https://crackstation.net/).
+
+Since attacks like rainbow tables exist, passwords need another layer of security.
```

Comment:
These two sentences say the same thing:
> Because [rainbow tables](https://en.wikipedia.org/wiki/Rainbow_table) exist, unfortunately.
> Since attacks like rainbow tables exist, passwords need another layer of security.

Get rid of the first?

Reply:
Sure, but this makes it so it doesn't fit the header of that section?
`### Why isn't hashing enough?`

Reply:
The whole section answers the question posed by the heading. The first sentence of the section need not be the direct answer. It's more appropriate for the last sentence to be the answer. 
Example:

>**Why isn't hashing enough?** Since the hashing function is easy to compute one-way, one possible strategy to break hashes is to build huge tables of plaintext and their corresponding hashes (these tables are called _rainbow tables_), so that cracking hashes becomes a simple problem of looking up the hash in the table. Because attacks like rainbow tables exist, hashing is not enough to store passwords securely.